File: BusInfo/app/src/main/java/com/daeseong/businfo/MainPagerAdapter.java
package com.daeseong.businfo;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

public class MainPagerAdapter extends FragmentStatePagerAdapter {

    private static final String TAG = MainPagerAdapter.class.getSimpleName();

    private String titles[] = {"나의경로", "버스조회", "즐겨찾기", "버스번호"};

    public MainPagerAdapter(FragmentManager fm) {
        super(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return new MainTab1Fragment();
            case 1:
                return new MainTab2Fragment();
            case 2:
                return new MainTab3Fragment();
            case 3:
                return new MainTab4Fragment();
            default:
                return null;
        }
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return titles[position];
    }

    @Override
    public int getCount() {
        return titles.length;
    }
}

File: BusInfo(kotlin)/app/src/main/java/com/daeseong/businfo/MainPagerAdapter.kt
package com.daeseong.businfo

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

class MainPagerAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    private val titles = arrayOf("나의경로", "버스조회", "즐겨찾기", "버스번호")

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> MainTab1Fragment()
            1 -> MainTab2Fragment()
            2 -> MainTab3Fragment()
            3 -> MainTab4Fragment()
            else -> throw IllegalArgumentException("Invalid position: $position")
        }
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return titles[position]
    }

    override fun getCount(): Int {
        return titles.size
    }
}
